Modules/System/Email/src/Email/EmailImpl.Codeunit.al

```diff
@@ -320,6 +320,50 @@ codeunit 8900 "Email Impl"
         until EmailRelatedRecord.Next() = 0;
     end;
 
+    procedure GetEmailOutboxForRecord(TableId: Integer; SystemId: Guid) ResultEmailOutbox: Record "Email Outbox" temporary;
+    begin
+        GetEmailOutboxForRecord(TableId, SystemId, ResultEmailOutbox);
+    end;
+
+    procedure GetEmailOutboxForRecord(TableId: Integer; SystemId: Guid; var ResultEmailOutbox: Record "Email Outbox" temporary)
+    var
+        EmailOutbox: Record "Email Outbox";
+        EmailRelatedRecord: Record "Email Related Record";
+        EmailAccountImpl: Codeunit "Email Account Impl.";
+    begin
+        EmailRelatedRecord.SetRange("Table Id", TableId);
+        EmailRelatedRecord.SetRange("System Id", SystemId);
+
+        if not EmailRelatedRecord.FindSet() then
+            exit;
+
+        if not EmailAccountImpl.IsUserEmailAdmin() then
+            EmailOutbox.SetRange("User Security Id", UserSecurityId());
+
+        repeat
+            EmailOutbox.SetCurrentKey("Message Id");
+            EmailOutbox.SetRange("Message Id", EmailRelatedRecord."Email Message Id");
+            if EmailOutbox.FindFirst() then begin
+                ResultEmailOutbox := EmailOutbox;
+                ResultEmailOutbox.Insert();
+            end;
+        until EmailRelatedRecord.Next() = 0;
+    end;
+
+    procedure GetOutboxEmailRecordStatus(MessageId: Guid) ResultStatus: Enum "Email Status"
+    var
+        EmailOutbox: Record "Email Outbox";
+        EmailAccountImpl: Codeunit "Email Account Impl.";
+    begin
+        if not EmailAccountImpl.IsUserEmailAdmin() then
+            EmailOutbox.SetRange("User Security Id", UserSecurityId());
+
+        EmailOutbox.SetCurrentKey("Message Id");
+        EmailOutbox.SetRange("Message Id", MessageId);
+        if EmailOutbox.FindFirst() then
```

Comment:
an error is more appropriate in case there is no Outbox Record

Reply:
I think it is better to keep same behaviour for both **GetSentEmailsForRecord** and **GetEmailOutboxForRecord** procedures.
Should I add errors for both procedures, keep them without error, or change only new one?

Reply:
It is not the same. In the cases above an empty list of records is returned, while here the first Email Status is returned which is not the same as the Outbox record does not exist/ or got deleted.

Reply:
sorry, understood it worng. I've removed "If Then" condition